libs/qCC_db/src/ccRasterGrid.cpp

```diff
@@ -266,133 +281,19 @@ bool ccRasterGrid::fillWith(	ccGenericPointCloud* cloud,
-						const ccColor::Rgb& col = cloud->getPointColor(n);
-						aCell.color = CCVector3d(col.r, col.g, col.b);
-					}
-				}
-			}
-
-			if (projectionType == PROJ_AVERAGE_VALUE)
-			{
-				//we keep track of the point which is the closest to the cell center (in 2D)
-				CCVector2d C = computeCellCenter(cellPos.x, cellPos.y, X, Y);
-				const CCVector3* Q = cloud->getPoint(aCell.pointIndex); //former closest point
-
-				CCVector2d P2D(P->u[X], P->u[Y]);
-				CCVector2d Q2D(Q->u[X], Q->u[Y]);
-
-				double squareDistToP = (C - P2D).norm2();
-				double squareDistToQ = (C - Q2D).norm2();
-				if (squareDistToP < squareDistToQ)
-				{
-					aCell.pointIndex = n;
-				}
-
-				if (hasColors)
-				{
-					assert(cloud->hasColors());
-					const ccColor::Rgb& col = cloud->getPointColor(n);
-					aCell.color += CCVector3d(col.r, col.g, col.b);
-				}
-
-			}
+			//If first point in cell, set head and tail to this reference
+			aCell.pointRefHead = pointRefList.data() + n;
+			aCell.pointRefTail = pointRefList.data() + n;
 		}
 		else
 		{
-			aCell.minHeight = aCell.maxHeight = P->u[Z];
-			aCell.pointIndex = n;
-
-			if (hasColors)
-			{
-				assert(cloud->hasColors());
-				const ccColor::Rgb& col = cloud->getPointColor(n);
-				aCell.color = CCVector3d(col.r, col.g, col.b);
-			}
-		}
-		
-		//sum the points heights
-		double Pz = P->u[Z];
-		aCell.avgHeight += Pz;
-		aCell.stdDevHeight += Pz * Pz;
-
-		//scalar fields
-		if (interpolateSF)
-		{
-			assert(pc);
-
-			//absolute position of the cell (e.g. in the 2D SF grid(s))
-			int pos = cellPos.y * static_cast<int>(width) + cellPos.x;
-			assert(pos < static_cast<int>(gridTotalSize));
-
-			for (size_t k = 0; k < scalarFields.size(); ++k)
-			{
-				assert(!scalarFields[k].empty());
-
-				CCCoreLib::ScalarField* sf = pc->getScalarField(static_cast<unsigned>(k));
-				assert(sf && pos < scalarFields[k].size());
-
-				ScalarType sfValue = sf->getValue(n);
-
-				if (ccScalarField::ValidValue(sfValue))
-				{
-					SF::value_type formerValue = scalarFields[k][pos];
-					if (aCell.nbPoints && std::isfinite(formerValue))
-					{
-						switch (sfInterpolation)
-						{
-						case PROJ_MINIMUM_VALUE:
-							// keep the minimum value
-							scalarFields[k][pos] = std::min<SF::value_type>(formerValue, sfValue);
-							break;
-						case PROJ_AVERAGE_VALUE:
-							//we sum all values (we will divide them later)
-							scalarFields[k][pos] += sfValue;
-							break;
-						case PROJ_MAXIMUM_VALUE:
-							// keep the maximum value
-							scalarFields[k][pos] = std::max<SF::value_type>(formerValue, sfValue);
-							break;
-						default:
-							assert(false);
-							break;
-						}
-					}
-					else
-					{
-						//for the first (valid) point, we simply have to store its SF value (in any case)
-						scalarFields[k][pos] = sfValue;
-					}
-				}
-			}
+			//Else point previous tail ref to this, and reset tail
+			*(aCell.pointRefTail) = pointRefList.data() + n;
```

Comment:
Why not 'pointRefList[n]` directly ?

Reply:
I am interested in the pointer to the pointRefList index n, not the data 
I guess it could also be written as 
&(pointRefList[n])